multiplatform-settings/src/appleMain/kotlin/com/russhwolf/settings/NSUbiquitousKeyValueStoreSettings.kt

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright 2020 Russell Wolf
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.russhwolf.settings
+
+import platform.Foundation.NSNotification
+import platform.Foundation.NSNotificationCenter
+import platform.Foundation.NSUbiquitousKeyValueStore
+import platform.Foundation.NSUbiquitousKeyValueStoreDidChangeExternallyNotification
+import platform.darwin.NSObjectProtocol
+
+/**
+ * A collection of storage-backed key-value data
+ *
+ * This class allows storage of values with the [Int], [Long], [String], [Float], [Double], or [Boolean] types, using a
+ * [String] reference as a key. Values will be persisted across app launches.
+ *
+ * The specific persistence mechanism is defined using a platform-specific implementation, so certain behavior may vary
+ * across platforms. In general, updates will be reflected immediately in-memory, but will be persisted to disk
+ * asynchronously.
+ *
+ * Operator extensions are defined in order to simplify usage. In addition, property delegates are provided for cleaner
+ * syntax and better type-safety when interacting with values stored in a `Settings` instance.
+ *
+ * On the iOS and macOS platforms, this class can be created by passing a [NSUbiquitousKeyValueStore] instance which will be used as a
+ * delegate, or via a [Factory].
+ */
+public class NSUbiquitousKeyValueStoreSettings public constructor(
+    private val delegate: NSUbiquitousKeyValueStore
```

Comment:
This can use the default value `NSUbiquitousKeyValueStore.defaultStore` since that will almost always be what users want.

```suggestion
    private val delegate: NSUbiquitousKeyValueStore = NSUbiquitousKeyValueStore.defaultStore
```